iib/workers/tasks/build.py

```diff
@@ -255,6 +255,50 @@ def get_rebuilt_image_pull_spec(request_id):
     )
 
 
+def _get_resolved_bundles(bundles):
+    """
+    Get the pull specification of the bundle images using their digests.
+
+    Determine if the pull spec refers to a manifest list.
+    If so, simply use the digest of the first item in the manifest list.
+    If not a manifest list, it must be a v2s2 image manifest and should be used as it is.
+
+    :param list bundles: the list of bundle images to be resolved.
+    :return: the list of bundle images resolved to their digests.
+    :rtype: list
+    :raises IIBError: if unable to resolve a bundle image.
+    """
+    log.info('Resolving bundles %s', ', '.join(bundles))
+    resolved_bundles = set()
+    for bundle_pull_spec in bundles:
+        skopeo_raw = skopeo_inspect(f'docker://{bundle_pull_spec}', '--raw')
+        if (
+            skopeo_raw.get('mediaType')
+            == 'application/vnd.docker.distribution.manifest.list.v2+json'
+        ):
+            # Get the digest of the first item in the manifest list
+            digest = skopeo_raw['manifests'][0]['digest']
+            if '@' in bundle_pull_spec:
+                repo = bundle_pull_spec.split('@', 1)[0]
```

Comment:
why do you get only first item? If it's expected to get amd64 (but question is still relevant, why only amd64), isn't safer to query via platform.architecture? What if there's going to be aarch64 in the future?

Reply:
As Luiz mentioned, it does not matter which arch manifest you get as long is it is not the manifest list digest. Hence, only the first item is taken.

Reply:
Yeah. The OLM team has explicitly requested this behavior. Operator bundle images are just data containers and are built for a particular arch only out of necessity.